server/query.go

```diff
@@ -127,7 +127,21 @@ func ParseQuery(query string) (pb.Query, error) {
 			bits = append(bits, bit)
 		}
 	}
-	out.Line = strings.Join(bits, "")
+
+	if len(bits) > 1 {
+		return out, errors.New("You cannot provide multiple of case:, lit:, and a bare regex")
+	}
+
+	if len(bits) > 0 {
+		out.Line = bits[0]
+	}
+
+	if out.Line == "" && out.File != "" {
```

Comment:
Just out of curiosity: Why did you decide put this logic here, vs pushing to the backend?

Mainly I'm trying to develop my intuition for the division of labor in the design :)

Reply:
I think my basic intuition is that I want to minimize arg-parsing or deep interpretation of the `Query` struct in `codesearch`, in place of keeping it as straightforward as possible an implementation of the RPC described in the proto, and I want `livegrep` to be the place that handles fiddly UX choices and the squishy problem of mapping between human inputs and something structured.

I'm not sure this is clearly right, since (e.g.) the best UX might mean pushing that squishiness all the way back, but that's my current approximate framework.

Reply:
Cool, that makes sense.